spec/models/audio_spec.rb

```diff
@@ -1,27 +1,84 @@
 require 'rails_helper'
 
 RSpec.describe Audio, type: :model do
-  before(:each) do
-    @uploader = User.create(uid: '123', nickname: 'louise')
-  end
+  it { is_expected.to belong_to(:uploader).class_name('User') }
+  it { is_expected.to validate_presence_of(:title) }
+  it { is_expected.to validate_length_of(:title).is_at_least(4) }
+  it { is_expected.to validate_presence_of(:filename) }
+
+  it do
+    create(:audio)
 
-  it "is not valid without a title"  do
-    audio = Audio.new(title: nil)
-    expect(audio).to_not be_valid
+    is_expected.to validate_uniqueness_of(:filename)
   end
 
-  it "is not valid when title is too short" do
-    audio = Audio.new(title: 'boo')
-    expect(audio).to_not be_valid
+  describe '#update_metadata' do
+    let(:flac) { double(:flac, url: 'flac-url') }
+    let(:he_aac) { double(:he_aac) }
+    let(:mp3_128) { double(:mp3_128) }
+    let(:file_hash) { { flac: flac, he_aac: he_aac, mp3_128: mp3_128 } }
+    let(:audio) { create(:audio) }
+
+    context 'when there is a flac file' do
+      before do
+        allow(audio).to receive(:file).and_return file_hash
+        allow(File).to receive(:exist?).with(flac.url).and_return true
+        allow(audio).to receive(:open).with("tmp/updates/#{audio.id}/#{audio.id}.flac", 'wb')
+        allow(File).to receive(:open).with("tmp/updates/#{audio.id}/#{audio.filename}.flac")
+        allow(File).to receive(:open).with("tmp/updates/#{audio.id}/#{audio.filename}.m4a")
+        allow(File).to receive(:open).with("tmp/updates/#{audio.id}/#{audio.filename}.mp3")
+        allow(flac).to receive(:replace)
+        allow(he_aac).to receive(:replace)
+        allow(mp3_128).to receive(:replace)
+      end
+
+      it 'transcodes the file' do
+        expect(audio).to receive(:transcode_updates).with("tmp/updates/#{audio.id}/#{audio.id}.flac")
+
+        audio.update_metadata
+      end
+
+      it 'saves the audio' do
+        allow(audio).to receive(:transcode_updates).with("tmp/updates/#{audio.id}/#{audio.id}.flac")
+
+        expect(audio).to receive(:save)
+
+        audio.update_metadata
+      end
+    end
+
+    context "when there isn't a flac file" do
+      before do
+        allow(audio).to receive(:file).and_return file_hash
+        allow(File).to receive(:exist?).with(flac.url).and_return false
+      end
+
+      it 'transcodes the file' do
```

Comment:
Is this test named correctly? When there isn't a flac file, it should *not* transcode the file.

Reply:
Ops, you are right.